src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs

```diff
@@ -384,6 +453,33 @@ public static void FileSystemWatcher_OnRenamed()
     }
 
     [Fact]
+    public static void FileSystemWatcher_OnRenameGivesExpectedFullPath()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var fsw = new FileSystemWatcher(dir.Path))
+        {
+            AutoResetEvent are = new AutoResetEvent(false);
+            string fullOriginalPath = Path.Combine(dir.Path, "Foo.txt");
+            string fullNewPath = Path.Combine(dir.Path, "Foo2.txt");
+
+            fsw.Renamed += (o, e) => 
+            {
+                Assert.Equal(e.OldFullPath, fullOriginalPath);
+                Assert.Equal(e.FullPath, fullNewPath);
+                are.Set();
+            };
+
+            fsw.EnableRaisingEvents = true;
+
+            using (var file = Utility.CreateTestFile(fullOriginalPath))
+            {
+                file.Move(fullNewPath);
+                are.WaitOne(Utility.WaitForExpectedEventTimeout);
+            }
+        }
+    }
+
```

Comment:
The comparisons being done are case-insensitive (for OS X).  Should there be a test that validates a mismatch in casing still fires the event(s)?